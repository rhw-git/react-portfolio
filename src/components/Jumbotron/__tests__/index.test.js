// import all library and package
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
// import component
import Jumbo from '..';
// delete the component created for testing
afterEach(cleanup);
//---------------tests-----------------//
describe('Jumbo component', () => {
  // renders Jumbo test
  it('render', () => {
    render(<Jumbo />);
  });
  // arrange
  const { asFragment } = render(<Jumbo />);
  // assert
  expect(asFragment()).toMatchSnapshot();
});

describe('emoji visibility', () => {
  it('inserts emoji into the ', () => {
    // arrange
    const { getByLabelText } = render(<Jumbo />);
    // assert
    expect(getByLabelText('wink')).toHaveTextContent('ğŸ˜‰');
  });
});
