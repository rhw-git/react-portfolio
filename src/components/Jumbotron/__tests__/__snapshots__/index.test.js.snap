// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<DocumentFragment>
  <div
    class="bg-dark text-white card"
    style="border-radius: 0;"
  >
    <div
      alt="Card image"
      style="background-image: url(jumbo-image.jpg); height: 16rem; width: 100%;"
      variant="top"
    />
    <div
      class="card-img-overlay"
    >
      <div
        class="card-title h5"
      >
        Hi, Welcome to My Personal Website !
      </div>
      <p
        class="card-text"
      >
        I am happy to assist you with your next website.
        <span
          aria-label="wink"
        >
          ðŸ˜‰
        </span>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
